MaxText/decode.py

```diff
@@ -173,71 +199,113 @@ def prefill_or_load(config, model, state, rng, sp_tokenizer, mesh, state_mesh_sh
       max_logging.log(f"Wrote prefill cache to {config.prefill_cache_dir}")
     return cache, last_logit, pos
 
-def decode_loop(config, state=None):
-  """Decoding loop for the Transformer model."""
-  rng = random.PRNGKey(0)
+def aqt_serving_conversion(model, mesh, config, model_vars, rng, sp_tokenizer):
+  """Compute the necessary prefill state"""
+
+  # Use configured prompt to generate default prompts
+  prompts = default_prompts(config)
+
+  # Compute shardings
+  replicated_sharding = jax.sharding.NamedSharding(mesh, P(None))
+  kv_cache_annotations = max_utils.get_kv_cache_annotations(model, config, rng, mesh)
+  kv_cache_mesh_shardings = jax.tree_map(
+      lambda p: jax.sharding.NamedSharding(mesh, p), kv_cache_annotations)
+
+  # Run prefill step
+  cache, logits, pos, new_vars = compute_prefill(
```

Comment:
Which variable fails to go through the code path at this point?

Reply:
We needed to call the "dummy" predict step after the "dummy" prefill step because it was initializing extra state for attention einsum. Discussed offline and decided to remove quantization in attention layer to keep the aqt initialization more light weight.